syntax = "proto3";

package proto;

service UsersInfo {
    rpc SignUp(SignUpInfo) returns (SignUpResponse);
    rpc SignIn(Credentials) returns (SignInResponse);
    // Logging
    rpc getLogFileReg(GetLogFileRequestReg) returns (stream GetLogFileReplyReg);
}

message SignUpInfo {
    bytes email = 1;
    bytes username = 2;
    bytes password = 3;
    bytes passwordConfirm = 4;
    bytes userType = 5;
    bytes airline = 6;
    bytes cartaDiCredito = 7;
}

message SignUpResponse {
    bool isOk = 1;
    string error = 2;
}

message Credentials {
    bytes username = 1;
    bytes password = 2;
}

message SignInResponse {
    bytes storedType = 1;
    bool isCorrect = 2;
}

message GetLogFileRequestReg {
    int32 numRichiesta = 1;
}



message GetLogFileReplyReg {
    bytes chunk_file  = 1;
    int32 num_chunk = 2;
}
